Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml

```diff
@@ -91,181 +91,13 @@ ParserQuery: |
-              '\x0d\x0a  <Data Name="RemoteMachineID">'RemoteMachineID:string'</Data>'*
-              | project-away EventData
-              | extend temp_isSrcMatch=has_any_ipv4_prefix(SrcIpAddr,src_or_any)
-               , temp_isDstMatch=has_any_ipv4_prefix(DstIpAddr,dst_or_any)
-              | extend ASimMatchingIpAddr = case(
-              array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-" // match not requested: probably most common case
-              , (temp_isSrcMatch and temp_isDstMatch), "Both" // has to be checked before the individual 
-              , temp_isSrcMatch, "SrcIpAddr"
-              , temp_isDstMatch, "DstIpAddr"
-              , "No match"
-              )
-             | where ASimMatchingIpAddr != "No match"
-             | project-away temp_*
-          ;
-      union SecurityEvent_5154_5155_5158_5159, SecurityEvent_5156_5157, SecurityEvent_5152
-      | lookup Directions on DirectionCode
-      | project-rename DvcHostname = Computer
-      | extend
-            SrcAppName = iff(isOutBound, Application, ""),
-            DstAppName = iff(not(isOutBound), Application, ""),
-            SrcDvcId = iff(isOutBound, RemoteMachineID, ""),
-            DstDvcId = iff(not(isOutBound), RemoteMachineID, ""),
-            SrcProcessId =  iff(isOutBound, tostring(ProcessId), ""),
-            DstProcessId =  iff(not(isOutBound), tostring(ProcessId), ""),
-            DstUserId = iff(isOutBound, RemoteUserID, ""),
-            SrcUserId = iff(not(isOutBound), RemoteUserID, ""),
-            DstHostname = iff(isOutBound, "", DvcHostname),
-            SrcHostname = iff(isOutBound, DvcHostname, "")
-          | project-away Application, RemoteMachineID, ProcessId, RemoteUserID
-            // *************** Postfilterring *****************************************************************
-          | where  (isnull(dstportnumber) or DstPortNumber == dstportnumber )
-          // *************** / Postfilterring *****************************************************************
-      };
   //////////////////////////////////////////////////////
   // this query extract the data from WindowsEvent table
   //////////////////////////////////////////////////////
+  let parser = (starttime:datetime=datetime(null), endtime:datetime=datetime(null)
+  , srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), ipaddr_has_any_prefix:dynamic=dynamic([]),dstportnumber:int=int(null)
+  , hostname_has_any:dynamic=dynamic([]), dvcaction:dynamic=dynamic([]),eventresult:string='*', disabled:bool=false
+   ) {
   let WindowsFirewall_WindowsEvent=(starttime:datetime=datetime(null), endtime:datetime=datetime(null)
   , srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), ipaddr_has_any_prefix:dynamic=dynamic([]),dstportnumber:int=int(null)
   , hostname_has_any:dynamic=dynamic([]), dvcaction:dynamic=dynamic([]),eventresult:string='*', disabled:bool=false
```

Comment:
I dont think we need to declare these filters twice

Reply:
done